SYCL/Basic/buffer/buffer_dev_to_dev.cpp

```diff
@@ -21,21 +23,35 @@ using namespace cl::sycl;
 int main() {
   int Data[10] = {-1, -1, -1, -1, -1, -1, -1, -1, -1, -1};
   {
-    buffer<int, 1> Buffer(Data, range<1>(10),
-                          {property::buffer::use_host_ptr()});
 
-    default_selector Selector;
+    const std::vector<platform> Platforms = platform::get_platforms();
+
+    if (Platforms.size() < 2) {
+      std::cout << "Need at least two platforms to create two separate "
```

Comment:
This claim does sound true. You can create separate context even with single device (a context to each of its subdevice). What exactly does the "context caching" mean?

Reply:
Context caching means that SYCL RT always returns the same context for a given platform.
PR which adds context caching https://github.com/intel/llvm/pull/2805.

>> You can create separate context even with single device (a context to each of its subdevice).

Yes, this is an option as well, but this requires that a device supports splitting to several sub-devices. Probably having such a device is more common than having two platforms on the system though.